public/app/plugins/datasource/elasticsearch/query_builder.js

```diff
@@ -243,19 +242,26 @@ function (queryDef) {
         }
       });
     }
-
+    var size = 500;
+    if(queryDef.size){
+      if(this.esVersion >= 5 && queryDef.size !== 0) {
```

Comment:
I believe in this case it is better to let ES complain in the case that the value is not supported, as this is an explicit definition by the user and by default it has a sane value.